### DISCUSSION TITLE
Execute SQL using jOOQ on arbitrary records?

### DISCUSSION THREAD

**Discussion by ghost** [jOOQ/jOOQ#15888] [Q&A] [answered]
Hello,

I see that there is an "import records into database" feature. But is there also a way of querying a list of records directly? The records could come from anywhere and I have an SQL expression / condition that I would like to execute against that set of records. It doesn't make much sense to import it into a database first. And designing my own expression language would be too much effort / pointless, since you've done all the hard work already.

Best regards

Andrin

**Comment by lukaseder** [comment, marked as answer]
You're essentially asking jOOQ to become an RDBMS? But why? There are very good RDBMS out there, including in-memory ones. And if your query is very simple, the `Stream` API is probably enough? And if it isn't, then maybe [jOOλ](https://github.com/jOOQ/jOOL) can help?

> And designing my own expression language would be too much effort / pointless, since you've done all the hard work already.

If you think the "hard work" in your question is designing the expression tree, and not the execution planning, then I have news for you 😅

**Reply by ghost** [reply]
OK. Thank you!
